src/utils/tsc.hpp

```diff
@@ -11,124 +11,34 @@
 
 #pragma once
 
-#include <sys/prctl.h>
-#include <x86intrin.h>
-
 #include <chrono>
+#include <functional>
 #include <optional>
 #include <thread>
 
+#include "utils/logging.hpp"
 #include "utils/timer.hpp"
```

Comment:
I think we don't need these, also `functional`,`thread` and `chrono` is not necessary.